nettacker/core/utils/ports.py

```
@@ -0,0 +1,49 @@
+        return []
+
+    ports = set()
+
+    for part in expression.split(","):
+        part = part.strip()
+
+        if "-" in part:
+            try:
+                start, end = map(int, part.split("-"))
+            except ValueError:
+                raise ValueError(f"Invalid port range: {part}")
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)